src/hooks/core/chat/useChatUtilsHook.ts

```diff
@@ -133,7 +135,32 @@ function useChatUtilsHook(): UseChatUtilsHook {
 
   const helperSaveChannelDetail = async (channel: ChannelList, response: any) => {
     if (!localDb) return;
-    const {originalMembers} = getChannelListInfo(response?.channel);
+    const builtChannelData = {
+      better_channel_member: response?.better_channel_members,
+      members: response?.members
+    };
+    const {
+      originalMembers,
+      anonUserInfoColorCode,
+      anonUserInfoColorName,
+      anonUserInfoEmojiCode,
+      anonUserInfoEmojiName,
+      channelImage,
+      channelName
+    } = getChannelListInfo(builtChannelData, signedProfileId, anonProfileId);
+
+    const channelData = await ChannelList.getSchemaById(localDb, channel?.id);
+
+    if (channelData !== null) {
+      channelData.anon_user_info_color_code = anonUserInfoColorCode;
+      channelData.anon_user_info_color_name = anonUserInfoColorName;
+      channelData.anon_user_info_emoji_code = anonUserInfoEmojiCode;
+      channelData.anon_user_info_emoji_name = anonUserInfoEmojiName;
+      channelData.channelPicture = channelImage;
+      channelData.name = channelName;
+
+      await channelData.save(localDb);
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This new hunk introduces additional properties to the `channelData` object. It's important to ensure that these new properties are properly validated before being assigned to avoid potential issues such as type errors or undefined values. If these properties are optional, consider providing default values.

```diff
+ const {
+   originalMembers,
+   anonUserInfoColorCode,
+   anonUserInfoColorName,
+   anonUserInfoEmojiCode,
+   anonUserInfoEmojiName,
+   channelImage,
+   channelName
+ } = getChannelListInfo(builtChannelData, signedProfileId, anonProfileId);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->